src/shared/cards/Card Template.js

```diff
@@ -0,0 +1,76 @@
+import Card from 'cards/Card';
+
+export default class CardName extends Card {
+  static title = 'Card Name';
+  cost = 4;
+  VP = 3;
+  coin = 2;
+  types = new Set(['Action', 'Treasure', 'Attack', 'Reaction', 'Victory']);
+  async onPlay(player) {
+    // Vanilla Effects
+    player.draw(1);
+    player.actions++;
+    player.buys++;
+    player.money++;
+
+    // Trash card(s) in Hand, is it game.currentPlayer, or just Player?
+    const cards = await game.currentPlayer.selectCards(min, max, predicate);
+    cards.forEach(card => {
+      player.trash(card);
+    })
+
+    // Trash this card
+    player.trash(this);
+
+    // Draw card and do something if()
+    card = player.deck.last();
+    if (card.types.has('Victory')) {  // look at top card in deck, discard if it is a victory card
+      player.moveCard(this, player.deck, player.discard);
+    }
+
+    // Draw multiple cards and do something if(), basically emulating library
+    const cards = player.setAside(min, max);
+    cards.forEach(card => {
+      if (card.types.has('Action')) {
+        const choice = await player.chooseOption(['hand', 'discard']);
+        if (choice === 'hard') {
+          player.moveCard(card, player.setAside, player.hand);
+        } else {
+          player.moveCard(card, player.setAside, player.discard);
+        }
+      }
+    });
+    player.unsetAside();
+
+
+    // Oponents draw card(s) and you do something , basically thief
+    await game.forEachOtherPlayer(async otherPlayers => {
+      const cards = otherPlayers.setAside(x,x);
+      const card = await otherPlayers.selectCards(1, 1, card => card.types.has(['Treasure'], cards);
```

Comment:
you can make that decision for them a lot of the time
```
if (cards.size > 1 && cards.every(card => card.types.has('Treasure')) {
  const card = await otherPlayer.selectCards(1, 1, cards);
  await otherPlayer.trash(card);
  await otherPlayer.discard(cards.find(c => c.id !== card.id));
} else {
  for (let i = 0; i < cards.length; i++) {
    const card = cards.get(i);
    if (card.types.has('Treasure')) {
      await otherPlayer.trash(card);
    } else {
      await otherPlayer.discard(card);
    }
  }
}
```

Reply:
You also don't need to use `aside` unless the card actually says put aside like in Library